fiftyone/utils/scale.py

```diff
@@ -0,0 +1,1077 @@
+    """
+    is_video = sample_collection.media_type == fomm.VIDEO
+
+    # Get label fields to export
+    label_fields = foc.get_label_fields(
+        sample_collection,
+        label_field=label_field,
+        label_prefix=label_prefix,
+        labels_dict=labels_dict,
+        required=False,
+        force_dict=True,
+    )
+
+    # Get frame label fields to export
+    if is_video:
+        frame_label_fields = foc.get_frame_labels_fields(
+            sample_collection,
+            frame_labels_field=frame_labels_field,
+            frame_labels_prefix=frame_labels_prefix,
+            frame_labels_dict=frame_labels_dict,
+            required=False,
+            force_dict=True,
+        )
+
+        if frame_label_fields and video_labels_dir is None:
+            raise ValueError(
+                "Must provide `video_labels_dir` when exporting frame labels "
+                "for video samples"
+            )
+
+    # Export the labels
+    labels = {}
+    with fou.ProgressBar() as pb:
+        for sample in pb(sample_collection):
+            # Compute metadata if necessary
+            if sample.metadata is None:
+                if is_video:
+                    metadata = fom.VideoMetadata.build_for(sample.filepath)
+                else:
+                    metadata = fom.ImageMetadata.build_for(sample.filepath)
+
+                sample.metadata = metadata
+                sample.save()
+
+            # Get frame size
+            if is_video:
+                frame_size = (
+                    sample.metadata.frame_width,
+                    sample.metadata.frame_height,
+                )
+            else:
+                frame_size = (sample.metadata.width, sample.metadata.height)
+
+            # Export sample-level labels
+            if label_fields:
+                labels_dict = _get_labels(sample, label_fields)
+                anno_dict = _to_scale_image_labels(labels_dict, frame_size)
+
+            # Export frame-level labels
+            if is_video and frame_label_fields:
+                frames = _get_frame_labels(sample, frame_label_fields)
+                if video_playback:
+                    annotations = _to_scale_video_playback_labels(
+                        frames, frame_size
+                    )
+                else:
+                    annotations = _to_scale_video_annotation_labels(
+                        frames, frame_size
+                    )
+
+                video_labels_path = os.path.join(
+                    video_labels_dir, sample.id + ".json"
+                )
+                etas.write_json(annotations, video_labels_path)
+                anno_dict = {"annotations": {"url": video_labels_path}}
+
+            labels[sample.id] = {
+                "filepath": sample.filepath,
+                "hypothesis": anno_dict,
+            }
+
+    etas.write_json(labels, json_path)
+
+
+def convert_scale_export_to_import(inpath, outpath):
```

Comment:
This is only for testing/demonstrative purposes, correct? Perhaps that could be explicitly noted in the doc string.

Reply:
Mostly, but it could be used to provide Importer/Exporter-like functionality for sharing labels in Scale format